### ISSUE TITLE
Equals method incorrectly compares timezone

### ISSUE THREAD

**Issue by tonyxiao** [moment/luxon#309]
According to [documentation](https://moment.github.io/luxon/docs/class/src/datetime.js~DateTime.html#instance-method-equals), 

![image](https://user-images.githubusercontent.com/696842/43368590-7aa48f06-9314-11e8-83f2-913a020a80b9.png)

However in practice
```typescript
const date = DateTime.fromISO('2018-01-01T22:00:00', { zone: 'America/Los_Angeles' })
const utcDate = DateTime.fromISO('2018-01-02T06:00:00', { zone: 'utc' })
expect(date.toMillis()).toEqual(utcDate.toMillis())
// The following two statement fails with error
// Expected value to equal: "2018-01-02T06:00:00.000Z", Received: "2018-01-01T22:00:00.000-08:00"
expect(date).toEqual(utcDate)
expect(date.equals(utcDate)).toEqual(true)
```

**Comment by icambron** [comment]
Yeah, the docstring is just wrong:

```js
  /**
   * Equality check
   * Two DateTimes are equal iff they represent the same millisecond
   * @param {DateTime} other - the other DateTime
   * @return {boolean}
   */
  equals(other) {
    return (
      this.isValid &&
      other.isValid &&
      this.valueOf() === other.valueOf() &&
      this.zone.equals(other.zone) &&
      this.loc.equals(other.loc)
    );
  }
```

**Comment by icambron** [comment]
Fixed in 5cb27b2
